main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"crypto/rsa"
+	"github.com/joho/godotenv"
+	"github.com/ufcg-lsd/arrebol-pb-worker/utils"
+	"github.com/ufcg-lsd/arrebol-pb-worker/worker"
+	"log"
+	"os"
+)
+
+const (
+	ConfFilePathKey = "CONF_FILE_PATH"
+	ServerEndpointKey = "SERVER_ENDPOINT"
+)
+
+func generateKeys(workerId string) {
+	log.Println("Starting to gen rsa key pair with workerid: " + workerId)
+	utils.GenAccessKeys(workerId)
+}
+
+//This functions sends the worker's public key to the server
+func sendKey(serverEndPoint string, workerId string) {
+	log.Println("Sending pub key to the server. ServerEndpoint: " + serverEndPoint)
+	url := serverEndPoint + "/workers/publicKey"
+	requestBody := &map[string]*rsa.PublicKey{"key": utils.GetPublicKey(workerId)}
+	httpResp := utils.Post(requestBody, url)
+
+	if httpResp.StatusCode != 201 {
+		log.Fatal("Unable to send public key to the server")
+	}
+}
+
+func isTokenValid(token string) bool {
+	return false
+}
+
+func main() {
+	// this main function start the worker following the chosen implementation
+	// passed by arg in the cli. The defaultWorker is started if no arg has been received.
+	err := godotenv.Load()
+
+	if err != nil {
+		log.Println("No .env file found")
+	}
+
+	switch len(os.Args) {
+	case 2:
+		workerImpl := os.Args[1]
+		println(workerImpl)
+	default:
+		defaultWorker()
+	}
+}
+
+func defaultWorker() {
+	// This is the default work behavior implementation.
+	// Its core stands for executing one task at a time.
+	log.Println("Starting reading configuration process")
+	file, err := os.Open(os.Getenv(ConfFilePathKey))
+
+	if err != nil {
+		log.Fatal("Error on opening configuration file", err.Error())
+	}
+
+	defer file.Close()
+
+	workerInstance := worker.ParseWorkerConfiguration(file)
+
+	serverEndpoint := os.Getenv(ServerEndpointKey)
+
+	//before join the server, the worker must send its public key
```

Comment:
every time this work starts it generates a new key? or it is a limitation of us not being able to read the key content from the file?

Reply:
yes, every time the worker starts, it generates a new key pair. We could check if it already exists before creating it. It can be useful in case of the worker process dies and someone restart it, but useless if the container dies... what do you think?

Reply:
I like checking. Be free to open an issue to be addressed later. Btw, I'm not sure making the worker container stateless is a good idea. Maybe, config, keys and stuff should use a filesystem mapped on host (this we can address latter too)